common/src/main/java/com/microsoft/identity/common/adal/internal/tokensharing/TokenShareUtility.java

```diff
@@ -99,6 +97,39 @@ public TokenShareUtility(@NonNull final String clientId,
         mTokenCache = cache;
     }
 
+    @NonNull
+    private static String getCanonicalEnvironment(@NonNull final String environment) {
+        switch (environment) {
+            case "login.windows.net":
+            case "login.microsoft.com":
+            case "sts.windows.net":
+                return ENVIRONMENT_GLOBAL;
+            case "login.chinacloudapi.cn":
+                return ENVIRONMENT_GALLATIN;
+            case "login.usgovcloudapi.net":
+                return ENVIRONMENT_ITAR;
+            default:
+                return environment;
```

Comment:
Should we be doing anything else to canonicalize this string?

Reply:
Should be sufficient